modules/custom/openy_upgrade_tool/src/ConfigEventIgnorePluginManager.php

```diff
@@ -0,0 +1,140 @@
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Plugin\DefaultPluginManager;
+
+/**
+ * Manages config event ignore plugins.
+ *
+ * @see \Drupal\openy_upgrade_tool\Annotation\ConfigEventIgnore
+ * @see \Drupal\openy_upgrade_tool\ConfigEventIgnoreBase
+ * @see \Drupal\openy_upgrade_tool\ConfigEventIgnorePluginManager
+ * @see plugin_api
+ */
+class ConfigEventIgnorePluginManager extends DefaultPluginManager {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(\Traversable $namespaces, CacheBackendInterface $cache_backend, ModuleHandlerInterface $module_handler) {
+    parent::__construct(
+      'Plugin/ConfigEventIgnore',
+      $namespaces,
+      $module_handler,
+      'Drupal\openy_upgrade_tool\ConfigEventIgnorePluginInterface',
+      'Drupal\openy_upgrade_tool\Annotation\ConfigEventIgnore'
+    );
+    $this->alterInfo('config_event_ignore_info');
+    $this->setCacheBackend($cache_backend, 'config_event_ignore');
+    $this->factory = new DefaultFactory($this->getDiscovery());
+  }
+
+  /**
+   * Get ignore rules for specified config type.
+   *
+   * @param string $config_type
+   *   Config type (entity_form_display, field_config, view, etc).
+   *
+   * @return array
+   *   Config ignore rules.
+   */
+  public function getPluginForConfigType($config_type) {
+    $plugins = [];
+
+    foreach ($this->getDefinitions() as $plugin_id => $plugin) {
+      try {
+        $instance = $this->createInstance($plugin_id);
+      }
+      catch (PluginException $exception) {
+        continue;
+      }
+
+      if ($instance->getConfigType() !== $config_type) {
+        continue;
+      }
+      $plugins[] = [
+        'plugin' => $instance,
+        'weight' => $instance->weight(),
+      ];
+    }
+
+    if (empty($plugins)) {
+      return NULL;
+    }
+    usort($plugins, ['\Drupal\Component\Utility\SortArray', 'sortByWeightElement']);
+    $data = reset($plugins);
+
+    return $data['plugin'];
+  }
+
+  /**
+   * Check config ignore rules.
+   *
+   * @param \Drupal\Core\Config\Config $config
+   *   Config object.
+   * @param $config_type
+   *   Config type.
+   *
+   * @return bool
+   *   TRUE - if changes can be ignored.
+   */
+  public function validateChanges(Config $config, $config_type) {
+    $original = $config->getOriginal();
+    $updated = $config->get();
+    $ignore_plugin = $this->getPluginForConfigType($config_type);
+
+    if (!$ignore_plugin) {
+      // Ignore rules not found for this config type.
+      return FALSE;
+    }
+
+    if ($ignore_plugin->fullIgnore()) {
+      // Ignore any changes for this config type.
+      return TRUE;
+    }
+
+    // Get config changes.
+    $diff = DiffArray::diffAssocRecursive($original, $updated);
+    $diff_keys = $this->getModifiedKeys($diff);
+
+    return $ignore_plugin->checkIgnoreDiffKeys($diff_keys);
+  }
+
+  public function getModifiedKeys(array $diff) {
```

Comment:
all methods should have dockblocks
Please, follow https://www.drupal.org/docs/develop/standards